src/main/java/org/codehaus/plexus/util/ReflectionUtils.java

```diff
@@ -218,6 +218,13 @@ private static void gatherVariablesAndValuesIncludingSuperclasses( Object object
 
         Class<?> clazz = object.getClass();
 
+        if (Class.class.getCanonicalName().equals(clazz.getCanonicalName()))
```

Comment:
Shouldn't we do this only on Java 16?

Reply:
Yes good catch (if Java >= 16 ... It is probable it doesn't change in the future).
I will check the way todo that (*System properties* `java.version` or other, `Runtime.version()`, ... ; with a correct backward compatibility, until Java 1.4 ; I have not prior version any-more ^^)

Reply:
> until Java 1.4

Until Java 1.7, due to `maven.compiler.source` Maven property for *plexus-utils* v3.4 (inherited from plexus parent)

Reply:
Proposal for change (`Runtime.version()` is only from Java 9): 
```
if (Integer.parseInt(System.getProperty("java.specification.version").split("\\.")[0]) >= 16 && Class.class.getCanonicalName().equals(clazz.getCanonicalName()))
```

----

@mkarg : With Java 11, there is (logically) this warning at execution:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.plexus.util.ReflectionUtils (file:/somewhere/git/plexus-utils/target/classes/) to field java.lang.Class.ANNOTATION
WARNING: Please consider reporting this to the maintainers of org.codehaus.plexus.util.ReflectionUtils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

I hesitate to include Java 11 in condition to avoid this warning 🤔. WDYT ?

Reply:
> I hesitate to include Java 11 in condition to avoid this warning

A warning is a "mini error" ^^ => v11 included in condition